backend/bittan/bittan/models/swish_payment_request.py

```diff
@@ -0,0 +1,73 @@
+from django.db import models
+from django_enumfield import enum
+import logging
+
+class PaymentStatus(enum.Enum):
+	# Seems to be duplicate of transaction declined, either here or in PaymentErrorCode. See https://github.com/Fysiksektionen/bittan/issues/13
+	""" Swish-side status of a payment """
+	PAID = 1
+	CANCELLED = 2 
+	CREATED = 3
+	DECLINED = 4
+	ERROR = 5
+
+	__SWISH_API_STATUS_MAPPINGS = {
+		"PAID": PAID,
+		"CANCELLED": CANCELLED,
+		"CREATED":CREATED,
+		"DECLINED": DECLINED,
+		"ERROR": ERROR,
+	}
+
+	@staticmethod
+	def from_swish_api_status(status: str):
+		# Try to get the status from the mapping otherwise log an error and None
+		if status not in PaymentStatus.__SWISH_API_STATUS_MAPPINGS:
+			logging.ERROR(f"Unknown Swish API status: {status}")
+			return None
+		return PaymentStatus[status]
+
+class PaymentErrorCode(enum.Enum):
+	UNKNOWN = 0
+	FAILED_TO_INITIATE = 1
+
+	TIMEOUT = 2
+	CANCELLED = 3
+
+	SWISH_HAS_NO_IDEA_WHAT_IS_HAPPENING = 4
+
+	
+	__SWISH_ERROR_CODE_MAPPINGS = {
+		"RF07": CANCELLED,
+		"BANKIDCL": CANCELLED,
+		"TM01": TIMEOUT,
+
+		"DS24": SWISH_HAS_NO_IDEA_WHAT_IS_HAPPENING,  # Make sure that this is handled! 
```

Comment:
Has this been handled? Either remove this comment or create an issue

Reply:
30c773aed991ad1f03f5a5f6c753c89d36f58dc0